mm2src/coins/nft/nft_structs.rs

```diff
@@ -129,13 +133,41 @@ impl fmt::Display for ContractType {
 }
 
 #[derive(Clone, Debug, Default, Deserialize, Serialize)]
-pub(crate) struct UriMeta {
+pub(crate) struct UriMetaFromStr {
     pub(crate) image: Option<String>,
+    /// the same as image, some tokens use **image_url** name for **image** field
+    /// but sometimes token_uri contains both`image_url` and `image`
+    pub(crate) image_url: Option<String>,
     #[serde(rename(deserialize = "name"))]
     pub(crate) token_name: Option<String>,
-    description: Option<String>,
-    attributes: Option<Json>,
-    animation_url: Option<String>,
+    pub(crate) description: Option<String>,
+    pub(crate) attributes: Option<Json>,
+    pub(crate) animation_url: Option<String>,
+    pub(crate) external_url: Option<String>,
+    pub(crate) image_details: Option<Json>,
+}
+
+#[derive(Clone, Debug, Default, Deserialize, Serialize)]
+pub(crate) struct UriMeta {
+    pub(crate) image: Option<String>,
+    pub(crate) token_name: Option<String>,
+    pub(crate) description: Option<String>,
+    pub(crate) attributes: Option<Json>,
+    pub(crate) animation_url: Option<String>,
+    pub(crate) external_url: Option<String>,
+    pub(crate) image_details: Option<Json>,
+}
+
+impl UriMeta {
+    pub(crate) fn merge_from(&mut self, other: UriMetaFromStr) {
+        self.image = self.image.clone().or(other.image).or(other.image_url);
+        self.token_name = self.token_name.clone().or(other.token_name);
+        self.description = self.description.clone().or(other.description);
+        self.attributes = self.attributes.clone().or(other.attributes);
+        self.animation_url = self.animation_url.clone().or(other.animation_url);
+        self.external_url = self.external_url.clone().or(other.external_url);
+        self.image_details = self.image_details.clone().or(other.image_details);
+    }
```

Comment:
I think we can do without cloning anything here e.g 
```rust
 if self.image == None {
            self.image = other.image
        }
```

Reply:
well yeah, I just thought that this version is less concise
![image](https://github.com/KomodoPlatform/atomicDEX-API/assets/52405288/38d1b806-8671-4ba2-a8b4-e386811b2949)

But if you think that `if` variant is more readable, I dont mind to change it.
Hmm and yeah, even if the increase of memory usage is temporary, too frequent allocation and deallocation of memory is bad, if there a lot of Some()

Reply:
Done